### PR TITLE
Fix pbs not correctly displaying

### PR DESCRIPTION
**Pull request by BenVardy**
Rebuilding the data base now also updates results as well as runners

### CODE REVIEW

**Comment by markvardy** [comment]
@BenVardy what testing have you done? We could test on a copy of the database and verify by running the UI locally against that copy.

Of course ideally we need actual automated tests. I've opened an issue for this.

File: src/event-input/calcPb.js [hunk 1]
```diff
@@ -13,26 +16,25 @@ const moment = require('moment');
...
-    let distance = result.distance;
+    let distance = result.distance.toString();
     if (!['2', '5'].includes(distance)) throw new Error('distance invalid');
 
     let fastest = runner.stats[`records${distance}k`].fastest;
     let slowest = runner.stats[`records${distance}k`].slowest;
 
 
-    if (!fastest) {
+    if (!fastest /* Therefore also !slowest */) {
```
Comment (ajdaniel) [inline src/event-input/calcPb.js]: This comment confuses me. I think you should place it on the next line. I'm guessing you're saying:
```
// There isn't a fastest time, thus there is no slowest time, so set them both
```

File: src/event-input/calcPb.js [hunk 2]
```diff
@@ -13,26 +16,25 @@ const moment = require('moment');
...
         runner.stats[`records${distance}k`].fastest = result;
-    } else if (!slowest) {
-        runner.stats[`records${distance}k`].slowest = result;
+        runner.stats[`records${distance}k`].slowest = result;
```
Comment (ajdaniel) [inline src/event-input/calcPb.js]: We should start moving towards "functional" programming whereby we do not alter any of the parameters in the function. Altering the parameters changes the variables outside of the function's scope and can have unintended side-effects. I know that you pass back the changed runner at the end of this function, but it's a good practice to adopt whereby you do not modify the parameter, just pass back a new version of the runner. You can do this either by cloning the data at the top of the function, or when you pass it back, pass back a merge of the old and the new e.g.:

```js
return {
	runner: {
		...runnerFromParameter,
		stats: newStats
	}
};
```

File: src/event-input/calcPb.js [hunk 3]
```diff
@@ -43,9 +45,8 @@ const calcPb = (result, runner) => {
     let bestAgeGrade = runner.stats.recordsAgeGrade.best;
     let worstAgeGrade = runner.stats.recordsAgeGrade.worst;
 
-    if (!bestAgeGrade) {
+    if (!bestAgeGrade /* Therefore also !worstAgeGrade */) {
```
Comment (ajdaniel) [inline src/event-input/calcPb.js]: Same as the previous comment about your comment, perhaps you could explain a bit better here. It almost seems like you're saying if(!bestAgeGrade) === if (!worstAgeGrade) which may also be true, but let's remove ambiguity

**Review by ajdaniel — CHANGES_REQUESTED** [review]
I agree with @markvardy can we run some tests here

**Review by ajdaniel — APPROVED** [review]
Looks good to me!

**Comment by ajdaniel** [comment]
@BenVardy I've approved your changes. Sorry it took me so long. I was gonna merge it but I just wanna check you're happy to do that, in case you were planning on doing tests.

**Comment by markvardy** [comment]
Ben & discussed it, manual testing has been done already.

**Pull request merged**